js/src/yoastseo-woo-identifiers.js

```diff
@@ -58,6 +71,26 @@ function doAllVariantsHaveIdentifier() {
 	return allVariantsHaveIdentifier;
 }
 
+/**
+ * A function to calculate whether or not all variants have at least one identifier set.
+ *
+ * @returns {Boolean} Whether or not all variants have at least one identifier set.
+ */
+function doAllVariantsHaveSkus() {
+	if ( ! hasVariants() ) {
+		return false;
+	}
+
+	// Gather all identifier objects for each variation. Make sure each has at least one non-empty identifier.
+	return skuStore.available_variations.every(
+		availableVariationId => {
+			const variation = skuStore.variations[ availableVariationId ];
+			// Return true if at least one identifier is set.
+			return Object.values( variation ).some( variationIdentifier => variationIdentifier !== "" );
```

Comment:
```suggestion
		availableVariationId => {
			const variationSku = skuStore.variations[ availableVariationId ];
			// Return true if variationSku is a nonempty string
			return typeof variationSku === "string" && variationSku.length > 0;
```

Something like this might work.